src/main/scala/firrtl/Utils.scala

```diff
@@ -691,6 +691,22 @@ object Utils extends LazyLogging {
     "SYNTHESIS",
     "PRINTF_COND",
     "VCS")
+
+  /** Expand a name into its prefixes, e.g., 'foo_bar__baz' becomes 'Seq[foo_, foo_bar__, foo_bar__baz]'. This can be used
+    * to produce better names when generating prefix unique names.
+    * @param name a signal name
+    * @param prefixDelim a prefix delimiter (default is "_")
+    * @return the signal name and any prefixes
+    */
+  def expandPrefixes(name: String, prefixDelim: String = "_"): Seq[String] = {
+    val regex = s"$prefixDelim[A-Za-z\\$$]".r
```

Comment:
Is there ever a case where we need to escape prefixDelim? e.g. if I set `prefixDelim = ?`

Reply:
This is a very good question...

Valid FIRRTL names, as far as I understand, are the same as Verilog, i.e., `[A-Za-z_][$A-Za-z_0-9]*`. However, if the user is trying to be clever they can have an identity transform that only computes some metadata and wants to track information using flattening, lowering, etc., but do it with a non-standard identifier. This is, with high probability, an anti-pattern, but it's allowable.

Due to what ClockList is doing, this should at least support `$`. If it works for that, then it should work for any input, I expect. _This currently doesn't split correctly on `$`..._

Do you have any thoughts on how this could work?

The `prefixDelim` should be, reasonably, any `String` that will unconditionally be viewed as a `String`. The user shouldn't be passing in regex. It's possible that this could be limited to a `Char`.

Reply:
It's fine if we don't support arbitrary delimiters here, but then we should document this very clearly since it could lead to surprises otherwise.

Reply:
I wasn't aware of `Regex.quote`. That took care of it. Also, I updated this to treat multiple copies of the delimiter as contributing to one prefix. This would allow for shenanigans with likely unique delimiters, e.g., "FOOBAR" could be a delimiter. I'll hold off on merging pending the discussion below.